src/Campaign.ts

```diff
@@ -0,0 +1,165 @@
+import i18next from 'i18next'
+import { CorruptionLoadout, type Corruptions } from './Corruptions'
+import { player } from './Synergism'
+import { DOMCacheGetOrSet } from './Cache/DOM'
+import { IconSets } from './Themes'
+
+export type AscensionModifiers = 'GlobalSpeed'
+
+export type CampaignLoadout = Partial<Corruptions>
+export type CampaignModifiers = Partial<Record<AscensionModifiers, number>>
+
+export type CampaignKeys = 'test1' | 'test2' | 'test3'
+
+export interface ICampaignManagerData {
+  currentCampaign?: CampaignKeys | undefined
+  campaigns?: Record<CampaignKeys, number>
+}
+
+export interface ICampaignData {
+  campaignCorruptions: Partial<Corruptions>
+  campaignModifiers: CampaignModifiers
+  limit: number
+  isMeta: boolean
+}
+
+export class CampaignManager {
+  private totalCampaignTokens: number
+  private currentCampaign: Campaign | undefined
+  private campaigns!: Record<CampaignKeys, Campaign>
+
+  constructor (campaignManagerData?: ICampaignManagerData) {
+    this.campaigns = {} as Record<CampaignKeys, Campaign>
+    if (campaignManagerData !== undefined) {
+      for (const campaignKey of Object.keys(campaignDatas)) {
+        const key = campaignKey as keyof typeof campaignDatas
+        this.campaigns[key] = new Campaign(campaignDatas[key], key, campaignManagerData.campaigns?.[key])
+      }
+      const currentKey = campaignManagerData.currentCampaign
+      if (currentKey !== undefined) {
+        this.currentCampaign = this.campaigns[currentKey]
+        player.corruptions.used = new CorruptionLoadout(this.currentCampaign.campaignCorruptions)
+      } else {
+        this.currentCampaign = undefined
+      }
+    }
+    else {
+      for (const campaignKey of Object.keys(campaignDatas)) {
+        const key = campaignKey as keyof typeof campaignDatas
+        this.campaigns[key] = new Campaign(campaignDatas[key], key, 0)
+      }
+      this.currentCampaign = undefined
+    }
+    this.totalCampaignTokens = this.computeTotalCampaignTokens()
+  }
+
+  computeTotalCampaignTokens = () => {
+    let sum = 0
+    for (const campaign in this.campaigns) {
+      const key = campaign as keyof typeof this.campaigns
+      sum += this.campaigns[key].computeTokenValue()
+    }
+    return sum
+  }
+
+  get tokens () {
+    return this.totalCampaignTokens
+  }
+
+  get current () {
+    return this.currentCampaign
+  }
+
+  // Store as this in player
+  get c10Completions (): Record<CampaignKeys, number> {
+    return Object.fromEntries(
+      Object.entries(this.campaigns).map(([key, value]) => [key, value.c10Completions])
+    ) as Record<CampaignKeys, number>
+  }
+}
+
+export class Campaign {
+  // Stored as variable out of scope
+  name: string
+  description: string
+  campaignCorruptions: CampaignLoadout
+  campaignModifiers: CampaignModifiers
+  limit: number
+  isMeta: boolean
+
+  // Saved as a variable
+  _c10Completions = 0
```

Comment:
1. the comment is meaningless. It won't make sense to anyone within a month.
2. why the underscore property name? Use a [private property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties)